### PR TITLE
Error #1136 fixed

### PR DESCRIPTION
**Pull request by ragreener1**
Stopped using distinct on columns as causes errors if switching from SQLite to MySQL
## Description

Very minor fix to solve bug when changing database - possible performance impact but not significanr
## Motivation and Context

This fixed error #1136 
## How Has This Been Tested?

Tested - fix worked as expected
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by justsomedudeonthenet** [comment]
I don't see any appreciable speed difference between the two select statements...in fact, they should return identical results.

Why does this fix the issue?

**Comment by averylarry** [comment]
I don't think anyone ever figured out why it didn't work correctly.  If I remember correctly it was discovered by trial-and-error that the group by syntax appeared to return correct results.  I don't even know if the results were verified.

My suspicion, though, is that using distinct on 2 fields isn't doing what people think it should.  Group by using 2 fields should work the way people think it should.

**Comment by justsomedudeonthenet** [comment]
Wait...its doing distinct on 2 fields?  I didn't catch that.  Which fields?

**Comment by tiagomartines11** [comment]
`GROUP BY` returns the expected results, it only groups whatever is specified on the GROUP BY clause.
`DISTINCT` returns wrong results, as it only considers as duplicate rows those that have all the same data (e.g., a duplicate spawnpoint_id might show up because they have different lat/log due to rounding mistakes)

For MySQL, the distinct query was introduce based on two reasons:
1) MySQL 5.7+ will complain/fail to execute that query, because it enforces sql_mode=only_full_group_by (newbie users had difficulties to disable/understand that).
2) It may have significant performance impact depending on your setup.

I believe we should stick to the DISTINCT query somehow, it seems more appropriate given that we are not doing any aggregation operations. I suggested an alternative to this issue on #1075, but that only tackles the lat/lon situation.

**Comment by averylarry** [comment]
Let's define disappear seconds as the number of seconds past the hour that a given spawnpoint disappears.  (disappear time's minutes * 60 plus disappear time's seconds).
In my data, many spawnpoints have multiple disappear seconds.  I'm not talking about multiples of 15 minutes either (for the 2x15, 30, 45, 60 spawns).  Some of them are only a second or 2 apart.
So the first obstacle -- that does NOT seem correct.
But moving on -- the entire premise of the select statement seems wrong to me (noting fully that I don't know any coding languages but I am decent with SQL queries).  It is supposed to return lat, lon, disappear seconds.  Now lat,lon is unique to each spawnpoint_id.  But disappear seconds isn't.  From a somewhat pure SQL syntax I would do this:
select latitude, longitude, disappear seconds
from pokemon
group by latitude,longitude,spawnpoint_id,disappear seconds

In SQL, this query is invalid:
select latitude,longitude,disappear seconds
from pokemon
group by spawnpoint_id

and so is this:
select latitude,longitude,disappear seconds, distinct(spawnpoint_id)
from pokemon

So I don't fully understand how sqlite and mysql decide what data to throw away.  Nor do I understand why we're throwing away data.
Still noting that I might totally misunderstand the code.

**Comment by averylarry** [comment]
#1075 is a separate issue.  That is bad (or at least inconsistent) data in the database, which (in my opinion) should not be fixed by changing how we query the database.  The bad data should be fixed.

**Comment by justsomedudeonthenet** [comment]
Has anyone tested this on MySQL 5.7, with ONLY_FULL_GROUP_BY turned on?

**Comment by beccasafan** [comment]
The duplicates are currently needed due to the way the map handles (well, doesn't handle) the [non-15m spawns](https://www.reddit.com/r/pokemongodev/comments/4yzqc2/spawn_point_types_clearing_confusion/).

When a spawn comes back with an invalid TTH, the code forces that down to 15m and this is why there are multiple times for the same spawnpoint.

Let's use a 1x30 spawnpoint as an example. This spawn is active from 0m-30m. It has a valid TTH from 15-30m.

If you scan this spawnpoint between 15m and 30m, it will have a valid TTH and every instance will have the same mm:ss disappear time.

However, if you scan this spawnpoint between 0m and 15m, it will have an invalid TTH. Let's say this scan happens at 2m. The code will force that TTH to 15m and thus the pokemon is magically active from 2m until 17m.

The spawnpoint now has times from 2m to 17m, and from 15m to 30m in the database.

If you remove the duplicates, then spawnpoint scanning might pick up that location for 2m to 17. Or it might pick it up from 15m to 30m. It's non-deterministic.

The "duplicates" are kept here so that when looking at the spawnpoint data and doing spawnpoint scanning, that spawnpoint will be scanned at these multiple times.

For another way to handle it, check out the get_spawnpoints method. This is the method used to get the data when you turn on the spawnpoint circles in the ui, and the data that shows in the info window when you hover/click on a spawnpoint circle. The logic here does handle the multiple times for a single spawnpoint. The logic in this method will return a single time for every spawnpoint by figuring out which one is the "real" one - which one has been seen the most. Since everything in the final 15m will have the same spawn time, that works out to be the "chosen" one.

If you hover over a spawnpoint that isn't a regular 15m spawn, it says "May appear as early as ...". This is because with the current handling of these other types, we know it's different, but we don't know what type it is. Still a cop out, but it's something.

Long story short, we kinda need the "duplicates" right now because they kind of aren't duplicates and we want to scan that point at the expanded times to look for a spawned pokemon.

*Whew!*

**Comment by sebastienvercammen** [comment]
We're cleaning up some old PRs that have been left around from before we had the time to handle everything properly. With the last few weeks going fantastically well, cleaning up the PR and issue queue will let us move forward again :) If this PR is still valid, don't hesitate to re-open and poke us.
